File: LeetCode/Difficulty/Easy/js/563.binary-tree-tilt.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var findTilt = function(root) {
    let sumOfDifference = 0;

    function findDifference(node){
        if(!node)
            return 0;
        if(node.val==null)
            return 0;

        const leftSum = findDifference(node.left);
        const rightSum = findDifference(node.right);

        sumOfDifference += Math.abs(leftSum - rightSum);

        return node.val + leftSum + rightSum;
    }

    findDifference(root);
    return sumOfDifference
};

File: LeetCode/Tag/Depth-first-Search/cpp/563.binary-tree-tilt.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
    int totalTilt = 0;

    int getSum(TreeNode* root) {
        if (!root) return 0;
        int leftSum = getSum(root->left);
        int rightSum = getSum(root->right);
        totalTilt += abs(leftSum - rightSum);
        return leftSum + root->val + rightSum;
    }

public:
    int findTilt(TreeNode* root) {
        getSum(root);
        return totalTilt;
    }
};